authors/apps/profiles/views.py

```diff
@@ -67,3 +70,125 @@ class AuthorsAPIView(ListView):
     queryset = User.objects.all()
     serializer_class = UserSerializer
     renderer_classes = (UserJSONRenderer,)
+
+
+class FollowsView(APIView):
+    permission_classes = (permissions.IsAuthenticated,)
+
+    def post(self, request, username):
+        """
+        User follows an Author
+        """
+        follower_id = User.objects.get(username=request.user.username).id
+        try:
+            followed_id = User.objects.get(username=username).id
+            self.profile_id = Profile.objects.get(user_id=followed_id).id
+            self.check_follow_status(
+                follower_id, followed_id, username)
+        except Exception as e:
+            if isinstance(e, User.DoesNotExist):
+                raise NotFound(
+                    'No user with name {} exists.'.format(username))
+            return Response(
+                {'error': str(e)}, status=status.HTTP_400_BAD_REQUEST)
+        follow_data = {'follower': follower_id, 'followed': self.profile_id}
+        serializer = FollowSerializer(data=follow_data)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        profile = self.get_profile_to_follow(self.profile_id)
+        return Response(profile, status=status.HTTP_201_CREATED)
+
+    def check_follow_status(self, follower_id, followed_id, name):
+        if follower_id == followed_id:
+            raise Exception('You cannot follow your own profile.')
+        query_result = Follow.objects.filter(
+            follower_id=follower_id, followed_id=self.profile_id)
+        if len(query_result) != 0:
+            raise Exception('You are already following {}.'.format(name))
+
+    def get_profile_to_follow(self, followed):
+        profile = Profile.objects.get(id=followed)
+        serializer = ProfileSerializer(profile)
+        profile = serializer.data
+        profile['following'] = True
```

Comment:
Thanks for the good work @joelethan, however, help clarify what you do in this function
. Do you follow a user? do u return the profile of the user one has followed?
I think the naming and the implementation are not clear enough.,